v4/newrelic/transaction.go

```diff
@@ -260,6 +304,7 @@ func (txn *Transaction) NewGoroutine() *Transaction {
 	newTxn := *txn
 	newTxn.thread = &thread{
 		currentSpan: txn.thread.currentSpan,
+		spanCount:   1,
```

Comment:
I feel like this number should match that on the `txn.thread`.

Reply:
Here's an example app that I think demonstrates my thinking:

```go
package main

import (
	"log"
	"net/http"
	"time"

	"github.com/newrelic/go-agent/v4/newrelic"
	"go.opentelemetry.io/otel/exporters/stdout"
)

func main() {
	pusher, err := stdout.InstallNewPipeline(nil, nil)
	if err != nil {
		log.Fatal(err)
	}
	defer pusher.Stop()

	app, _ := newrelic.NewApplication(
		newrelic.ConfigFromEnvironment(),
	)
	txn := app.StartTransaction("main")
	defer txn.End()

	txn.StartSegment("AAAAAAAAAAAAAAAAAA").End()

	go func(txn *newrelic.Transaction) {
		hdr := http.Header{}
		hdr.Set("traceparent", "00-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-bbbbbbbbbbbbbbbb-01")
		txn.AcceptDistributedTraceHeaders("HTTP", hdr)
		txn.StartSegment("CCCCCCCCCCCCCCCCCC").End()
	}(txn.NewGoroutine())

	time.Sleep(time.Second)

	txn.StartSegment("BBBBBBBBBBBBBBBBBB").End()
}
```

As currently implemented, the Accept call will swap out the root span and therefore segment C will have a different trace ID than A and B.

Reply:
You're right. I fixed this and added a test.